src/Security/Authorization/Core/src/DenyAnonymousAuthorizationRequirement.cs

```diff
@@ -9,6 +9,7 @@ namespace Microsoft.AspNetCore.Authorization.Infrastructure;
 /// <summary>
 /// Implements an <see cref="IAuthorizationHandler"/> and <see cref="IAuthorizationRequirement"/>
 /// which requires the current user must be authenticated.
+/// Unlike what the name of the class implies, where you would expect failure if not authenticated, it instead succeeds if you are authenticated. 
```

Comment:
```suggestion
/// This calls <see cref="AuthorizationHandlerContext.Succeed"/> for authenticated users. Like all built-in requirements, it never calls <see cref="AuthorizationHandlerContext.Fail"/>. The <see cref="DefaultAuthorizationEvaluator"/> produces a failed <see cref="AuthorizationResult" /> when any requirement has not succeeded even if other requirements have succeeded, and no requirement has failed.
```

I'm basing this off the conversation in https://github.com/dotnet/aspnetcore/issues/35610. @blowdart @Tratcher Does this look good to you?

Reply:
Wow, that makes a lot more sense 😆